typescript/introduction/tsc-variable-declaration/tsc-destructuring.md

```diff
@@ -0,0 +1,116 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Destructuring
+
+---
+## Content
+
+Destructuring is a method used to unpack values (from arrays and tuples) or properties (from objects). It makes the code more readable by omitting some steps behind the scenes.
+
+Let's take a look at an example:
+
+```ts
+let numbers: number[] = [
+  1,
+  2
+];
+
+// this code
+let a = numbers[0];
+let b = numbers[1];
+
+// can be abbreviated to
+let [a, b] = numbers;
+
+console.log(a); // 1
+console.log(b); // 2
+```
+
+If you want to extract the remaining elements of the array into a single variable, you can use the rest operator (`...`):
+
+```ts
+let alphabet: string[] = [
+  'a',
+  'b',
+  'c',
+  'd'
+];
+
+let [first, ...rest] = alphabet;
+console.log(first); // a
+console.log(rest);
+// ['b', 'c', 'd']
```

Comment:
Maybe saying "rest operator" and using the `rest` variable name can confuse people as to which is which

```suggestion
let [first, ...others] = alphabet;
console.log(first); // a
console.log(others);
// ['b', 'c', 'd']
```